pkg/sql/catalog/tabledesc/validate.go

```diff
@@ -675,13 +675,31 @@ func (desc *wrapper) ValidateSelf(vea catalog.ValidationErrorAccumulator) {
 
 	// Validate mutations and exit early if any of these are deeply corrupted.
 	{
+		var mutationIDs util.FastIntSet
 		mutationsHaveErrs := false
 		for _, m := range desc.Mutations {
+			mutationIDs.Add(int(m.MutationID))
 			if err := validateMutation(&m); err != nil {
 				vea.Report(err)
 				mutationsHaveErrs = true
 			}
 		}
+		jobIDs := make(map[descpb.MutationID]catpb.JobID)
+		for _, mj := range desc.MutationJobs {
+			if !mutationIDs.Contains(int(mj.MutationID)) {
+				vea.Report(errors.AssertionFailedf("unknown mutation ID %d associated with job ID %d",
+					mj.MutationID, mj.JobID))
+				mutationsHaveErrs = true
+				continue
+			}
+			if jobID, found := jobIDs[mj.MutationID]; found {
+				vea.Report(errors.AssertionFailedf("duplicate job mappings %d and %d for mutation ID %d",
+					jobID, mj.JobID, mj.MutationID))
```

Comment:
nit: can the error message be easier to understand, something like "two jobs exists for the same mutation ID"? 

TIL: one mutation ID corresponds to one or more mutations on the mutation slice (i.e. they all share the same mutation ID as a result of the same DDL statement), but one mutation ID always corresponds to one job ID.